src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -80,8 +98,39 @@ def __init__(self, start: int, end: int):
         )
         self._condition = threading.Condition()
         self._no_remote_peer = 0
+        self._metadata_keys = [key.lower() for key in metadata_keys]
+        self._include_all_metadata = "*" in [
+            key.strip() for key in metadata_keys
+        ]
+        self._metadata_by_peer = collections.defaultdict(
+            messages_pb2.LoadBalancerStatsResponse.MetadataByPeer
+        )
 
-    def on_rpc_complete(self, request_id: int, peer: str, method: str) -> None:
+    def _add_metadata(
+        self,
+        rpc_metadata: messages_pb2.LoadBalancerStatsResponse.RpcMetadata,
+        metadata: MetadataType,
```

Comment:
Hm, should we disambiguate the arg names here? Maybe `rpc_metadata` and `metadata_to_add`? Or `src_metadata` and `dst_metadata`?